python/lsst/dax/imgserv/getimage/imagegetter.py

```diff
@@ -0,0 +1,406 @@
+        id_type, keyvals = self._get_dataID_from_qr(qresults)
+        if id_type == "RCFF": # rcff=run, camcol, field, filter
+            run, camcol, field, filtername = keyvals
+            log.debug("_get_imagefrombutler run={} camcol={} field={} \
+                    filter={}".format(run, 
+                camcol, field, filtername))
+            img = self._butler.get(self._imagedataset_type, run=run, camcol=camcol,
+                    field=field, filter=filtername)
+            return img 
+        elif id_type == "TPF": # tpf=tract, patch, filtername
+            tract, patch, filtername = keyvals
+            self._log.debug("deepCoadd _get_imagefrombutler tract={} patch={} \
+                    filtername={}".format(tract, patch, filtername))
+            img = self._butler.get(self._imagedataset_type, tract=tract,
+                    patch=patch, filter=filtername)
+            return img
+
+    def _get_imagedatasetMD(self):
+        #Return the butler policy name to retrieve metadata
+        return self._imagedataset_type + "_md"
+
+    def get_dataID_from_request(self, request):
+        """Returns the data ID, a dictionary of key value pairs, from the request with
+        valid being true if there were entries for everything in butler keys.
+        This will not work for floating point values.
+
+        """
+        valid = True
+        ids = {}
+        for key in self._butler_keys:
+            value = request.args.get(key)
+            if value is None:
+                valid = False
+            try:
+                value = int(value)
+            except:
+                value = str(value)
+            ids[key] = value
+        return ids, valid
+
+    def get_imagebydataID(self, data_id):
+        """Retrieve and image from the butler by the image id values in the dictionary ids
+        The needed values are specified in _butler_keys.
+        
+        """
+        img = self._butler.get(self._imagedataset_type, dataId=data_id)
+        return img 
+
+    def get_fullimage(self, ra, dec, filtername):
+        """Return an image containing ra and dec with filtername (optional)
+        Returns None if no image is found.
+        This function assumes the entire image is valid. (no overscan, etc.)
+        
+        """
+        img, metadata = self.get_fullimage_w_metadata(ra, dec, filtername)
+        return img
+
+    def get_fullimage_w_metadata(self, ra, dec, filtername):
+        """Return an image containing ra, dec, and filtername (optional) with corresponding metadata.
+        Returns None if no image is found.
+        This function assumes the entire image is valid. (no overscan, etc.)
+        
+        """
+        res = self._find_nearestimage_containing(ra, dec, filtername)
+        img = self._get_imagefrombutler(res)
+        metadata = self._get_metadata(res)
+        return img, metadata
+
+    def get_image(self, ra, dec, filtername, width, height, cutout_type="arcsecond"):
```

Comment:
Using image_cutout considering data id is implicit, not explicit as through the parameters.